source/common/stream_info/stream_info_impl.h

```diff
@@ -231,6 +231,16 @@ struct StreamInfoImpl : public StreamInfo {
     return virtual_cluster_name_;
   }
 
+  void setDeferredLoggingHeadersAndTrailers(
```

Comment:
Can DeferredLoggingHeadersAndTrailers object lives in response decoder implementation which should be accessible by HCM::ActiveStream. The response decoder implementation also handles deferred logging, so there can be less indirection.

Reply:
Done.